libcxx/include/__expected/expected.h

```diff
@@ -1165,7 +1165,7 @@ class expected : private __expected_base<_Tp, _Err> {
   _LIBCPP_HIDE_FROM_ABI friend constexpr bool operator==(const expected& __x, const _T2& __v)
 #  if _LIBCPP_STD_VER >= 26
     requires(!__is_std_expected<_T2>::value) && requires {
-      { *__x == __v } -> __core_convertible_to<bool>;
+      { *__x == __v }->__core_convertible_to<bool>;
```

Comment:
There're still pure formatting changes. Let's drop them from this PR.
```suggestion
      { *__x == __v } -> __core_convertible_to<bool>;
```